pkg/e2etests/metrics_e2e_test.go

```diff
@@ -0,0 +1,1186 @@
+	allTimeSeries, metricNames, _, _ := GetTestMetricsData(startTimestamp)
+
+	err := initTestConfig()
+	assert.Nil(t, err)
+
+	err = ingestTestMetricsData(allTimeSeries)
+	assert.Nil(t, err)
+
+	mSegs, err := rotateMetricsDataAndClearSegStore(true)
+	assert.Nil(t, err)
+	assert.Greater(t, len(mSegs), 0)
+
+	err = initializeMetricsMetaData()
+	assert.Nil(t, err)
+
+	timeRange := &dtypeutils.MetricsTimeRange{
+		StartEpochSec: uint32(startTimestamp),
+		EndEpochSec:   uint32(startTimestamp + 4600),
+	}
+	expectedResults := map[string][]float64{
+		"testmetric0": {25, 50, 60, 70},
+		"testmetric1": {50, 90, 100, 110},
+		"testmetric2": {75, 130, 140, 150},
+	}
+
+	for _, mName := range metricNames {
+
+		query := fmt.Sprintf("(%v)", mName)
+		metricQueryRequest, _, _, err := promql.ConvertPromQLToMetricsQuery(query, timeRange.StartEpochSec, timeRange.EndEpochSec, 0)
+		assert.Nil(t, err)
+
+		res := segment.ExecuteMetricsQuery(&metricQueryRequest[0].MetricsQuery, &metricQueryRequest[0].TimeRange, 0)
+		assert.NotNil(t, res)
+		assert.Equal(t, 1, len(res.Results))
+
+		seriesDp := res.Results[fmt.Sprintf("%v{", mName)]
+		assert.NotNil(t, seriesDp)
+		assert.Greater(t, len(seriesDp), 0)
+
+		seriesDpValues := make([]float64, 0)
+		for _, dp := range seriesDp {
+			seriesDpValues = append(seriesDpValues, dp)
+		}
+		sort.Slice(seriesDpValues, func(i, j int) bool {
+			return seriesDpValues[i] < seriesDpValues[j]
+		})
+
+		assert.EqualValues(t, expectedResults[mName], seriesDpValues)
+	}
+}
+
+func Test_SimpleMetricQuery_Regex_on_MetricName_Star(t *testing.T) {
+	defer cleanUp(t)
+
+	startTimestamp := dataStartTimestamp
+	allTimeSeries, _, _, _ := GetTestMetricsData(startTimestamp)
+
+	err := initTestConfig()
+	assert.Nil(t, err)
+
+	err = ingestTestMetricsData(allTimeSeries)
+	assert.Nil(t, err)
+
+	mSegs, err := rotateMetricsDataAndClearSegStore(true)
+	assert.Nil(t, err)
+	assert.Greater(t, len(mSegs), 0)
+
+	err = initializeMetricsMetaData()
+	assert.Nil(t, err)
+
+	timeRange := &dtypeutils.MetricsTimeRange{
+		StartEpochSec: uint32(startTimestamp),
+		EndEpochSec:   uint32(startTimestamp + 4600),
+	}
+
+	intervalSeconds, err := mresults.CalculateInterval(timeRange.EndEpochSec - timeRange.StartEpochSec)
+	assert.Nil(t, err)
+	assert.Equal(t, uint32(20), intervalSeconds)
+
+	query := `({__name__=~"testmetric.*"})`
+	metricQueryRequest, _, _, err := promql.ConvertPromQLToMetricsQuery(query, timeRange.StartEpochSec, timeRange.EndEpochSec, 0)
+	assert.Nil(t, err)
+
+	/*
+		Expected Results:
+		At T1: testmetric0: 10, testmetric1: 20, testmetric2: 30
+		At T2: testmetric0: 40, testmetric1: 80, testmetric2: 120
+		At T3: testmetric0: 50, testmetric1: 90, testmetric2: 130
+		At T4: testmetric0: 60, testmetric1: 100, testmetric2: 140
+		At T5: testmetric0: 70, testmetric1: 110, testmetric2: 150
+
+		The time diff between T1 and T2 is less than 6 seconds, so the values at T1 and T2 will be aggregated to the same bucket.
```

Comment:
Why is 6 seconds important? Where is that coming from?

Reply:
Actually I need to update this comment, I will update these comments.